From 172a47b616241a97935f2655927a3d2a9efe61bc Mon Sep 17 00:00:00 2001
From: zlong2 <zhifang.long@intel.com>
Date: Tue, 22 May 2018 14:44:33 +0800
Subject: [PATCH 5/7] minigbm : fix black screen on some machines

Cause : on those machines, display connector type_id is 3,
	while old logic only consider 1 and 2.
Solution : try type_id 3 if 1 and 2 failed.

Note : just temp solution, the candidate connector
       search logic needs to be refactored to avoid
       similar issue.

Change-Id: I52d6ea228506bf1d22fbb2e1ba68bdc39cd015c2
Tracked-On:
Signed-off-by: zlong2 <zhifang.long@intel.com>
Reviewed-on: 630020
Reviewed-by: Zhang, Hongyu <hongyu.zhang@intel.com>
Tested-by: Zhang, Hongyu <hongyu.zhang@intel.com>
Reviewed-on: 674562
Tested-by: Xue, Yifei <yifei.xue@intel.com>
Reviewed-by: Shi, Zhongjie <zhongjie.shi@intel.com>
---
 drv.c | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/drv.c b/drv.c
index 0f68fb7..903ac6a 100755
--- a/drv.c
+++ b/drv.c
@@ -1178,14 +1178,21 @@ int drv_init_kms(struct driver* drv)
 
     edp = fetch_connector(kms, DRM_MODE_CONNECTOR_DisplayPort, type_id);
     if(!edp) {
-        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_DisplayPort, type_id+1);        
-    }    
+        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_DisplayPort, type_id+1);
+    }
+    if(!edp) {
+        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_DisplayPort, type_id+2);
+    }
+
     if(!edp) {
-        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_eDP, type_id);        
-    }    
+        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_eDP, type_id);
+    }
+    if(!edp) {
+        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_eDP, type_id+1);
+    }
     if(!edp) {
-        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_eDP, type_id+1);        
-    }        
+        edp = fetch_connector(kms, DRM_MODE_CONNECTOR_eDP, type_id+2);
+    }
 
     if (edp) {
         kms->edp_available = 1;
-- 
2.21.0

